src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/PackageDeclarationCheck.java

```diff
@@ -140,11 +140,12 @@ public void beginTree(DetailAST ast) {
     @Override
     public void finishTree(DetailAST ast) {
         if (!defined) {
-            int lineNumber = DEFAULT_LINE_NUMBER;
-            if (ast != null) {
-                lineNumber = ast.getLineNo();
+            if (ast == null) {
+                log(DEFAULT_LINE_NUMBER, MSG_KEY_MISSING);
```

Comment:
@romani ping. It seems to me we need to document this that the check isn't fully xpath compliant.

Reply:
yes, this is exception case.
unfortunately empty java files are valid files by javac:
```
/var/tmp$ cat Test.java 
/var/tmp$ javac Test.java 
/var/tmp$ echo $?
0
```
So we have two options:
1) do not report violation, as at https://checkstyle.org/config_coding.html#PackageDeclaration we all talk about classes, but file is compilation unit, file is not mean class. No classes/types - no violations.
2) update documentation of Check and Filter  that for such cases user need to use non Xpath suppression.

this code come to us at https://github.com/checkstyle/checkstyle/commit/2ce3fc4fd0ee5b731cd86459a46db89c085e4c19 , as you can see in issue there was no demand for such files to have violation, there was problem with NPE - https://github.com/checkstyle/checkstyle/issues/1149 .

I suggesting do "2)" now , create separate issue to fix "1)" and do it separately.

@rnveach , if you agree, let do this.

Reply:
I agree. @DXTkastb please make the change

Reply:
I am ok to loose this  validation on empty files. Nobody requested it, value of it is very small.

Reply:
I have updated the documentation. If anything needs to be changed or added to the documentation please let me know.